packages/astro/src/core/render/dev/css.ts

```diff
@@ -22,9 +22,9 @@ export function getStylesForURL(filePath: URL, viteServer: vite.ViteDevServer):
 
 	// recursively crawl module graph to get all style files imported by parent id
 	function crawlCSS(id: string, scanned = new Set<string>()) {
-		// note: use .idToModuleMap() for lookups (.urlToModuleMap() may produce different
-		// URLs for modules depending on conditions, making resolution difficult)
-		const moduleName = viteServer.moduleGraph.idToModuleMap.get(id);
+		// note: use .urlToModuleMap() for lookups to ensure up-to-date module info
+		// using .idToModuleMap() could cause missing style imports on initial server load
+		const moduleName = viteServer.moduleGraph.urlToModuleMap.get(id);
```

Comment:
I wonder if we are meant to be using `moduleGraph.getModuleById()` or `moduleGraph.getModuleByUrl()` or `moduleGraph.getModuleByFile()`. That might do some handling for us, vs. direct access like this.

Reply:
Lasly, there is a `viteID()` function that we've used internally for things like this

Reply:
oh, and of course Vite has it's own `{normalizePath}` export 😅

Reply:
I was _about_ to explain why `getModuleByUrl` doesnt' work with our `viteID`... then you mentioned `getModulesByFile`. Looks like this is the best of both worlds!
- Gets the module for our ID
- Gets any related modules with the same file URL as our ID (this is where our missing tailwind import is found)

Curious how this affects performance, but I modified the PR to use `getModulesByFile` instead. An alternative may be 1. try to use `getModuleById`, then 2. try to use `getModuleByUrl` if no `importedModules` are found.